### ISSUE TITLE
Xamarin.iOS

### ISSUE THREAD

**Issue by dodyg** [pgstath/Sharp.Xmpp#24]
Is there any fundamental issue that would make this library not working on iOS? I am just curious. Right now I am thinking of adopting Sharp.Xmpp for a messaging functionality for my Xamarin Android/iOS app.

**Comment by ilber** [comment]
I'm interested too... I want to use this library in a cross platform app and in iOS seems not working.. Any news? Will someone give us any idea of what is going wrong?

**Comment by pgstath** [comment]
Hi both.While it will compile and run I have not tested it in iOS. However, the fundamental problem is with iOS and XMPP: if its not an VoIP app iOS will tear down the background TCP connections (when the app is going to the background), thus messages will not be delivered. 
A solution would be to implement XEP 0365 Push Notifications, XEP-0365 currently not supported

**Comment by gviddy** [comment]
I would like to look into this and get it going on iOS. Where in the code would be the best place to start ?

**Comment by pgstath** [comment]
I do not think that you will have a problem to run it. The problem will be that when the application leaves the foreground, the TCP connnections in the background will be closed by the OS.

**Comment by ilber** [comment]
We have already started to integrate it in our app and it seems working fine. Of course you need to take care of the situation when the app goes in background but enabling archives and push on server side you will not loose any message. @pgstath thank you for your effort on creating this library.

**Comment by pgstath** [comment]
Thx, nice to hear good news like this!!

**Comment by daspuru** [comment]
Hello! I am trying to use Sharp.Xmpp in a Xamarin.iOS project. Added the NuGet Package, and then tried to use the code and I can not add any "using" statement that recognize the XmppClient class. 

@ilber , can you help me with this?

Thanks.

**Comment by ilber** [comment]
@daspuru I'm not using the package since there is no library for iOS or Android. I just created 2 Projects (iOS Library and Android Library) that link all the files from the main one. Then just reference this projects into your app.
